modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -641,6 +641,9 @@ public ObjectEntry deleteObjectEntry(ObjectEntry objectEntry)
 			objectEntry.getGroupId(), objectDefinition.getObjectDefinitionId(),
 			objectEntry.getPrimaryKey());
 
+		_objectEntryVersionLocalService.deleteObjectEntryVersion(
+			objectEntry.getObjectEntryId());
+
```

Comment:
This method deletes the object entry version using its own ID (objectEntryVersionId), not the object entry ID